src/mantine-core/src/Floating/FloatingArrow/get-arrow-position-styles.ts

```diff
@@ -90,7 +88,7 @@ export function getArrowPositionStyles({
     [radiusByFloatingSide[side]]: rem(arrowRadius),
   };
 
-  const arrowPlacement = withBorder ? -arrowSize / 2 - 1 : -arrowSize / 2;
+  const arrowPlacement = rem(-arrowSize / 2);
```

Comment:
While playing with big REM units on the Storybook to check if the arrow was positioned correctly, I found that the -1 px offset was in fact unnecessary, even at base REM value of 16px. It added an imperceptible gap that became more and more visible with bigger REM values, already quite visible at REM = 20px.

So I removed it, and consequently removed the `withBorder` prop altogether since it did nothing else than that.

Reply:
Here at REM = 32px:
<img width="534" alt="image" src="https://user-images.githubusercontent.com/12484011/222997540-d3a3c716-5091-4eb4-a15c-40e9d1b91753.png">

Behavior if we keep the -1 offset:
<img width="534" alt="image" src="https://user-images.githubusercontent.com/12484011/222997609-6685bd6d-e6fe-4a82-abf1-55774fa9d87d.png">